### PR TITLE
Inline Commenting: Optimize store selector and misc changes

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
PR updates the `CollabSidebar` component and combines store subscriptions when possible. I've also removed the side effect for setting `blockCommentID`.

I noticed multiple `useSelect` hook usages when reviewing the #66583.

## Why?
* When possible, combine the same store selectors instead of using different `useSelect` hooks. This reduces the number of store subscriptions.
* There's no need to synchronize data in the local state.

## Testing Instructions
1. Enable experimental inline commenting option under Gutenberg -> Experiments.
2. Confirm that the Inline Commenting works as before.

### Testing Instructions for Keyboard
Same.

### CODE REVIEW

File: packages/editor/src/components/collab-sidebar/index.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ import { store as editorStore } from '../../store';
 import AddCommentButton from './comment-button';
 import AddCommentToolbarButton from './comment-button-toolbar';
 
-const threadsEmptyArray = [];
+const EMPTY_ARRAY = [];
```
Comment (Mamaduka) [inline packages/editor/src/components/collab-sidebar/index.js:25]: This matches the name used in other places and makes it easier to search the code base.

File: packages/editor/src/components/collab-sidebar/index.js [hunk 2]
```diff
@@ -54,50 +54,41 @@ export default function CollabSidebar() {
...
+					status: 'any',
+					per_page: 100,
+			  } )
+			: null;
+
 		return {
-			postId: select( editorStore ).getCurrentPostId(),
+			postId: _postId,
+			postStatus: getEditedPostAttribute( 'status' ),
+			threads: data ?? EMPTY_ARRAY,
```
Comment (ellatrix) [inline packages/editor/src/components/collab-sidebar/index.js:75]: Nit: could also fall back to an array outside of the selector and we wouldn't need the constant.

**Review by ellatrix — APPROVED** [review]

File: packages/editor/src/components/collab-sidebar/index.js (hunk 2 above)
Comment (Mamaduka) [reply packages/editor/src/components/collab-sidebar/index.js:75]: Let's leave it for now. The component seems to change more rapidly now that the general concept has landed and we have the updated designs.

**Review by tyxla — APPROVED** [review]
Tests well and code makes sense 👍 Thanks @Mamaduka 🚀

**Pull request merged**